### ISSUE TITLE
Sveltekit | "moduleResolution": "Bundler" | Could not find a declaration file for module 'svelte-headless-table'

### ISSUE THREAD

**Issue by bfovez** [bryanmylee/svelte-headless-table#173]
Hi,

Recently, `npm create svelte@latest` changed its beahvior and now adds the the following in the `"compilerOptions"` section of `tsconfig.json`: `"moduleResolution": "Bundler"`.

Since then, typescript yields dozens of the following errors:
`Could not find a declaration file for module 'svelte-headless-table'. 'c:/Users/----/node_modules/svelte-headless-table/index.js' implicitly has an 'any' type.
  Try ``npm i --save-dev @types/svelte-headless-table`` if it exists or add a new declaration (.d.ts) file containing ``declare module 'svelte-headless-table';``ts(7016)`

We could remove `"moduleResolution": "Bundler"` from `tsconfig.json`, but:
- It's awkward for any new Sveltekit's projects/users.
- What if we actually need the `"moduleResolution": "Bundler"` clause?

Thank you.

**Comment by ohschway** [comment]
Just faced this problem and was coming here to open an issue about it.

**Comment by kuatroka** [comment]
+1

**Comment by pheuter** [comment]
Same issue now on the latest [SvelteKit 2](https://kit.svelte.dev/docs/migrating-to-sveltekit-2).

**Comment by shellwhale** [comment]
+1

**Comment by DrokeRavens** [comment]
+1

**Comment by tika** [comment]
How do you fix this?

**Comment by kuatroka** [comment]
wait for an update. It doesn't really break anything for me for now. Only an error message.

**Comment by nhuxhr** [comment]
# Resolving TypeScript Module Recognition Issue in Svelte Headless Table

I encountered this issue recently and managed to resolve it using the following temporary solution.

## The Problem

TypeScript wasn't recognizing the module. Upon investigating the package in the `./node_modules/svelte-headless-table` directory, I noticed that the types were located in the `./node_modules/svelte-headless-table/lib` folder. TypeScript seemed to be unhappy with this location for the types.

## Temporary Solution

> **Note:** I prefer placing my types in the `@types` folder.

1. Create a `@types` folder in the root of your project.
2. Create a `[NAME].d.ts` file in the `@types` folder (e.g., `svelte-headless-table.d.ts`).
3. Update your Svelte/SvelteKit TypeScript config `tsconfig.json` to include the `@types` folder.

```json5
{
  // ...

  "include": [
    // NOTE: This is the important part

    // If you want to overwrite includes/excludes, make sure to copy over the relevant includes/excludes
    // from the referenced tsconfig.json - TypeScript does not merge them in

    // "./.svelte-kit/tsconfig.json" include start
    "./.svelte-kit/ambient.d.ts",
    "./.svelte-kit/non-ambient.d.ts",
    "./.svelte-kit/types/**/$types.d.ts",
    "./vite.config.js",
    "./vite.config.ts",
    "./src/**/*.js",
    "./src/**/*.ts",
    "./src/**/*.svelte",
    "./tests/**/*.js",
    "./tests/**/*.ts",
    "./tests/**/*.svelte",
    // "./.svelte-kit/tsconfig.json" include end

    "./@types/*.d.ts" // <--- Add this line
  ]

  // ...
}
```

4. Add the following content to the `@types/[NAME].d.ts` file you created in step 2 (e.g., `@types/svelte-headless-table.d.ts`).

```ts
declare module "svelte-headless-table" {
  // Copy the types from the `./node_modules/svelte-headless-table/lib/index.d.ts` file and update the import paths from `./` to `./node_modules/svelte-headless-table/lib/`
  export * from "svelte-render";
  export { Subscribe } from "svelte-subscribe";
  export { createTable } from "./node_modules/svelte-headless-table/lib/createTable";
  export { Table } from "./node_modules/svelte-headless-table/lib/createTable";
  export { HeaderRow } from "./node_modules/svelte-headless-table/lib/headerRows";
  export {
    HeaderCell,
    FlatHeaderCell,
    DataHeaderCell,
    FlatDisplayHeaderCell,
    GroupHeaderCell,
    GroupDisplayHeaderCell,
  } from "./node_modules/svelte-headless-table/lib/headerCells";
  export {
    BodyRow,
    DisplayBodyRow,
    DataBodyRow,
  } from "./node_modules/svelte-headless-table/lib/bodyRows";
  export {
    BodyCell,
    DataBodyCell,
    DisplayBodyCell,
  } from "./node_modules/svelte-headless-table/lib/bodyCells";
  export {
    Column,
    FlatColumn,
    DataColumn,
    GroupColumn,
    DisplayColumn,
  } from "./node_modules/svelte-headless-table/lib/columns";
  export type {
    TableViewModel,
    TableState,
  } from "./node_modules/svelte-headless-table/lib/createViewModel";
  export type {
    DataLabel,
    DisplayLabel,
    HeaderLabel,
  } from "./node_modules/svelte-headless-table/lib/types/Label";
}
```

5. Restart the TypeScript server in your IDE, and you should be good to go.

> **Note:** This solution was quickly devised while working on a project, and I'm not certain if there is a better way to handle this. I'll update this answer if I discover a more optimal solution. I hope this helps someone out there.

**Comment by bryanmylee** [comment]
Fixed by #178 and #181. I'll be releasing a v0.18 to cover this.
